### PR TITLE
[GEOT-7055] Split gt-geojsonstore in two modules, gt-geojson-core and gt-geojson-store

### PR DESCRIPTION
**Pull request by aaime**
[GEOT-7055](https://osgeo-org.atlassian.net/browse/GEOT-7055) [Powered by Pull Request Badge](https://pullrequestbadge.com/?utm_medium=github&utm_source=geotools&utm_campaign=badge_info)

As per discussion during the last PSC meeting.
Also added jacoco support so that we can measure code coverage from maven (looks good).

One annoying bit is that the module changed name, which would break existing users.
On the other end, it makes the naming consistent with other cases of related modules, and it's the last good time to do such change, before the modules are proposed for supported status.

GeoServer follow up PR here:
https://github.com/geoserver/geoserver/pull/5549

# Checklist

- [x] I have read the [contribution guidelines](https://github.com/geotools/geotools/blob/main/CONTRIBUTING.md).
- [x] I have sent a [Contribution Licence Agreement](https://docs.geotools.org/latest/developer/procedures/contribution_license.html) (not required for small changes, e.g., fixing typos in documentation).
- [x] First PR targets the `main` branch (backports managed later; ignore for branch specific issues).
- [x] Avoid [Java 9+ split packages](http://tutorials.jenkov.com/java/modules.html#split-packages-not-allowed).
- [x] All the build checks are green ([see automated QA checks](https://docs.geotools.org/latest/developer/conventions/code/qa.html)).

For core and extension modules:

- [ ] New unit tests have been added covering the changes.
- [ ] [Documentation](https://github.com/geotools/geotools/tree/main/docs) has been updated (if change is visible to end users).
- [x] There is an issue in [GeoTools Jira](https://osgeo-org.atlassian.net/projects/GEOT) (except for changes not visible to end users). 
- [x] Commit message(s) must be in the form ``[GEOT-XYZW] Title of the Jira ticket``.
- [ ] Bug fixes and small new features are presented as a single commit.
- [x] The commit targets a single objective (if multiple focuses cannot be avoided, each one is in its own commit, and has a separate ticket describing it).

### CODE REVIEW

**Comment by aaime** [comment]
@ianturton thoughts?

File: modules/unsupported/geojson-core/src/test/java/org/geotools/data/geojson/GeoJSONTestSupport.java [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+
+    private static String removeWhitespace(String actual) {
+        if (actual == null) return "";
+        StringBuffer crush = new StringBuffer(actual);
+        int ch = 0;
+        while (ch < crush.length()) {
+            char chracter = crush.charAt(ch);
+            if (Character.isWhitespace(chracter)) {
+                crush.deleteCharAt(ch);
+                continue;
```
Comment (ianturton) [inline modules/unsupported/geojson-core/src/test/java/org/geotools/data/geojson/GeoJSONTestSupport.java:43]: minor nitpick: would an `else` be more efficient here?

**Review by ianturton — APPROVED** [review]
Technically, I would say the Jacoco changes should go in a different PR but provided you don't mind I doubt anyone else will :-)

File: modules/unsupported/geojson-core/src/test/java/org/geotools/data/geojson/GeoJSONTestSupport.java (hunk 1 above)
Comment (aaime) [reply modules/unsupported/geojson-core/src/test/java/org/geotools/data/geojson/GeoJSONTestSupport.java]: Not sure if it's going to be more efficient, but, updated it, and rebased the PR.
I'm going to run a full build on the geoserver side too, and if everything is green, merge.

**Comment by aaime** [comment]
Here goes nothing!

**Pull request merged**